docs: add info about self-hosted github/gitlab instances (#26501)
nuxt/nuxt: docs/2.guide/3.going-further/7.layers.md

If you want to extend a private remote source, you need to add the environment variable `GIGET_AUTH=<token>` to provide a token.
::

::tip

> If you want to extend a remote source from GitHub or GitLab instances, you need to add the environment variable `GIGET_GITHUB_URL=<url>` or `GIGET_GITLAB_URL=<url>` to provide a url.

Reviewer: ```suggestion
If you want to extend a remote source from a self-hosted GitHub or GitLab instance, you will need to supply its URL with an environment variable: `GIGET_GITHUB_URL=<url>` or `GIGET_GITLAB_URL=<url>`.
```

> If you want to extend a remote source from a self-hosted GitHub or GitLab instance, you will need to supply its URL with an environment variable: `GIGET_GITHUB_URL=<url>` or `GIGET_GITLAB_URL=<url>`.

Reviewer 2: ```suggestion
If you want to extend a remote source from a self-hosted GitHub or GitLab instance, you need to supply its URL with a `GIGET_GITHUB_URL=<url>` or `GIGET_GITLAB_URL=<url>` environment variable.
```

Reviewer 3: I think we might suggest [auth option](https://github.com/unjs/c12?tab=readme-ov-file#extending-config-layer-from-remote-sources).

I guess we documented it somewhere else in docs as well.

Reviewer: yes, the auth option is documented a few lines above...

Author: Auth option was mentioned [here](https://nuxt.com/docs/getting-started/layers#usage)
